src/controllers/recommendationController.js

```diff
@@ -0,0 +1,23 @@
+const AlbumRecommendation = require('../models/AlbumRecommendation');
+const recommendationController = require('../controllers/recommendationController');
+async function generateAlbumRecommendations(userId) {
+  return await AlbumRecommendation.find({ userId });
+}
+const getAlbumRecsForUser = async (req, res) => {
+  const { userId } = req.params;
+  try {
+    const recommendations =
+      await recommendationController.generateAlbumRecommendations(userId);
+    res.json(recommendations);
+  } catch (error) {
+    console.error('Error while generating recommendations:', error);
+    res
+      .status(500)
+      .json({ error: 'An error occurred while generating recommendations.' });
+  }
```

Comment:
Keep in mind, now that we've defined this route in `expressServer` _under_ all our middleware, that means any errors raised by this code will be handled by: https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/c6ca1551a1d1906a3d66bd33174b52f4c58bd98b/src/middleware/error-handler.js#L3-L2

Which means it is redundant to have the try/catch here (because the errors that occur here, will be caught by the middleware function and handled centrally in that function; instead of every controller code needing to be wrapped with try/catch). 

```suggestion
  const recommendations =
    await recommendationController.generateAlbumRecommendations(userId);
  return res.json({ recommendations: recommendations });
```

Reply:
And for following common convention in this app, we will also wrap our return value in an JSON object `{}` with the key `recommendations`.